cylc/flow/terminal.py

```diff
@@ -257,17 +265,13 @@ def wrapper(*api_args: str) -> None:
                         list(api_args),
                         **parser_kwargs
                     )
-                    if hasattr(opts, 'color'):
-                        use_color = (
-                            opts.color == 'always'
-                            or (opts.color == 'auto' and supports_color())
-                        )
+                    use_color = should_use_color(opts)
                     wrapped_args = [parser, opts, *args]
                 if 'color' in inspect.signature(wrapped_function).parameters:
                     wrapped_kwargs['color'] = use_color
 
                 # configure Cylc to use colour
-                color_init(autoreset=True, strip=not use_color)
+                color_init(autoreset=False, strip=not use_color)
```

Comment:
?

Reply:
The autoreset causes `ESC[0m` to be added at the beginning of the last line of log files when the scheduler is run daemonized. Don't know of how else to fix that

Reply:
Right, so this is the change that hides the bug.

Reply:
I'm not keen on this change (it protects users from us missing reset codes), but I also can't see an obvious way to avoid it, daemonization does strange things.

Reply:
Hopefully if we stick to using `ansimarkup.parse` with the xml-style tags it should be easy to avoid accidentally not resetting colour

Reply:
No, that does not protect us:

```python
>>> parse('<green>foo')
'\x1b[32mfoo'
```

Reply:
It doesn't guarantee anything, but it's obvious there should be a closing tag at least

Reply:
I take it (from PR approval) that you've agreed with this approach now @oliver-sanders ?

Reply:
Auto reset is a better approach, but I'm not going to hold this up on that point.

Reply:
Opened issue for re-enabling it: #6076